hermes-api/src/main/java/pl/allegro/tech/hermes/api/constraints/ContentTypeValidator.java

```diff
@@ -0,0 +1,21 @@
+package pl.allegro.tech.hermes.api.constraints;
+
+import pl.allegro.tech.hermes.api.ContentType;
+import pl.allegro.tech.hermes.api.DeliveryType;
+import pl.allegro.tech.hermes.api.Subscription;
+
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class ContentTypeValidator implements ConstraintValidator<ValidContentType, Subscription> {
+
+    @Override
+    public void initialize(ValidContentType constraintAnnotation) {
+
+    }
+
+    @Override
+    public boolean isValid(Subscription subscription, ConstraintValidatorContext context) {
+        return !(subscription.getDeliveryType() == DeliveryType.BATCH && subscription.getContentType() == ContentType.AVRO);
```

Comment:
I did this on purpose so if in the future there will be a new delivery/content type nothing breaks. If you still think it would be better then no problem.

Reply:
I mean, they are actually equal, but I think that the first version shows the intent better. 
But as I said, I do not object to change this.